sync/src/orphan_block_pool.rs

```diff
@@ -84,17 +88,50 @@ impl OrphanBlockPool {
                 .and_then(|blocks| blocks.get(hash).cloned())
         })
     }
+}
+
+// NOTE: Never use `LruCache` as container. We have to ensure synchronizing between
```

Comment:
Add another note that RwLock ensures the consistency between maps. Using multiple concurrent maps does not work here.